app/flows/initial_request_flow.rb

```diff
@@ -0,0 +1,218 @@
+        form_object: form_object,
+        provider: provider,
+      }
+    elsif pick_a_provider_page?
+      {
+        training_providers: training_providers_from_query,
+      }
+    else
+      {
+        training_providers: training_providers_without_associated,
+        form_object: form_object,
+        provider: provider,
+      }
+    end
+  end
+  # rubocop:enable Lint/DuplicateBranch: Duplicate branch body detected
+
+  def redirect?
+    return false if valid_number_of_places?
+
+    (training_provider_search? && one_search_result?) || training_provider_selected?
+  end
+
+  def allocation
+    if (training_provider_search? && one_search_result?) || training_provider_selected?
+
+      allocations.find_by(provider_code: training_provider[:provider_code])
+    end
+  end
+
+  def redirect_path
+    if allocation
+      edit_provider_recruitment_cycle_allocation_path(
+        provider.provider_code,
+        provider.recruitment_cycle_year,
+        training_provider.provider_code,
+        id: allocation[:id],
+      )
+    elsif (training_provider_search? && one_search_result?) || training_provider_selected?
+      initial_request_publish_provider_recruitment_cycle_allocations_path(
+        provider_code: provider.provider_code,
+        recruitment_cycle_year: recruitment_cycle.year,
+        training_provider_code: training_provider[:provider_code],
+      )
+    end
+  end
+
+  delegate :valid?, to: :form_object
+
+private
+
+  def valid_number_of_places?
+    params[:number_of_places] && /\A\d+\z/.match?(params[:number_of_places]) && params[:number_of_places].to_i.positive?
+  end
+
+  def number_of_places_zero?
+    params[:number_of_places]&.to_i&.zero?
+  end
+
+  def training_provider_selected?
+    params[:training_provider_code].present? && params[:training_provider_code] != "-1"
+  end
+
+  def training_provider_search?
+    params[:training_provider_code] == "-1" && params[:training_provider_query].present? && params[:training_provider_query].size > 1
+  end
+
+  def one_search_result?
+    training_providers_from_query.count == 1
+  end
+
+  def form_object
+    permitted_params = params.slice(:training_provider_code, :training_provider_query, :number_of_places)
+                             .permit(:training_provider_code, :training_provider_query, :number_of_places)
+
+    @form_object ||= Publish::InitialRequestForm.new(permitted_params)
+  end
+
+  def allocations
+    @allocations ||= Allocation.includes(:provider, :accredited_body)
+    .current_allocations.where(accredited_body_code: provider.provider_code)
+  end
+
+  def training_provider_search_service
+    API::V2::AccreditedProviderTrainingProvidersController::TrainingProviderSearch
+  end
+
+  def training_providers_with_fee_paying_pe_course
+    @training_providers_with_fee_paying_pe_course ||=
+
+      training_provider_search_service.new(
+        provider: provider, params: { filter: { subjects: PE_SUBJECT_CODE,
```

Comment:
This is some dodgy indentation

Reply:
Fixed, we really should invest in our linter through.